snapshot.go

```diff
@@ -511,8 +479,15 @@ func (es *EventuallyFileOnlySnapshot) NewIterWithContext(
 	if es.excised.Load() {
```

Comment:
We can remove this

Reply:
Kept it in because the work done to perform an atomic load is a lot less than performing an unnecessary iterator construction. Left it in for now.